oneargopy/Argo.py

```diff
@@ -0,0 +1,279 @@
+        self.download_settings = DownloadSettings(user_settings)
+        self.source_settings = SourceSettings(user_settings)
+        if self.download_settings.verbose: print(f'Starting initialize process...\n')
+        
+        if self.download_settings.verbose: print(f'Your current download settings are: {self.download_settings}')
+        if self.download_settings.verbose: print(f'Your current source settings are: {self.source_settings}')
+
+        # Check for and create subdirectories if needed
+        if self.download_settings.verbose: print(f'Checking for subdirectories...\n')
+        self.__initialize_subdirectories()
+
+        # Download files from GDAC to Index directory
+        if self.download_settings.verbose: print(f'\nDownloading index files...\n')
+        for file in self.download_settings.index_files:
+            self.__download_index_file(file)
+
+        # Load the argo_synthetic-profile_index.txt file into a data frame
+        if self.download_settings.verbose: print(f'\nTransfering index files into data frames...')
+        self.synthetic_index = self.__load_synthetic_dataframe()
+        self.prof_index = self.__load_prof_dataframe()
+
+        # Print number of floats
+        if self.download_settings.verbose: self.__display_floats() 
+
+        if self.download_settings.verbose: print(f'Initialize is finished!')
+
+        if not self.download_settings.keep_index_in_memory:
+            if self.download_settings.verbose: print('Removing dataframes from memory...')
+            del self.synthetic_index
+            del self.prof_index
+
+
+    def __initialize_subdirectories(self) -> None:
+        """ A function that checks for and creates the necessary folders as 
+            listed in the download settings sub_dir list. 
+        """
+        for directory in self.download_settings.sub_dirs:
+            directory_path = self.download_settings.base_dir.joinpath(directory)
+            if directory_path.exists():
+                if self.download_settings.verbose: print(f'The {directory_path} directory already exists!')
+            else:
+                try:
+                    if self.download_settings.verbose: print(f'Creating the {directory} directory!')
+                    directory_path.mkdir()
+                except Exception as e:
+                    if self.download_settings.verbose: print(f'Failed to create the {directory} directory: {e}')
+
+
+    def __download_index_file(self, file_name: str) -> None:
+        """ A function to download and save an index file from GDAC sources. 
+
+            :param: filename : str - The name of the file we are downloading.
+        """
+        index_directory = Path(self.download_settings.base_dir.joinpath("Index"))
+
+        # Get the expected filepath for the file
+        file_path = index_directory.joinpath(file_name)
+
+        # Check if the filepath exists
+        if file_path.exists():
+
+            # Check if the settings allow  for updates
```

Comment:
allow for updates

Reply:
![image](https://github.com/lovelysandlonelys/OneArgo-Python-Dev/assets/91633941/639826bd-ad45-451f-bcbe-9a9ffaa5b2f3)